controller.go

```diff
@@ -931,15 +932,15 @@ func parseAllocateConstraintsResponse(source interface{}, machine *machine) (Con
 	constraintsMap := valid["constraints_by_type"].(map[string]interface{})
 	result := ConstraintMatches{
 		Interfaces: make(map[string][]Interface),
-		Storage:    make(map[string][]BlockDevice),
+		Storage:    make(map[string][]StorageDevice),
 	}
 
 	if interfaceMatches, found := constraintsMap["interfaces"]; found {
 		matches := convertConstraintMatches(interfaceMatches)
```

Comment:
So we are processing the "interfaces" map here.
The "interfaces" schema defines a map of string-> int list. 
We should have a version of convertConstraintMatches() which returns the correct type so we don't need to use id.(int)